### PR TITLE
Send the UDP packet to all resolved IP addresses

### PR DESCRIPTION
**Pull request by dentarg**
If given IP address as host, do not try to resolve it.

### CODE REVIEW

File: src/sparoid.mjs [hunk 1]
```diff
@@ -74,15 +77,30 @@ export default class Sparoid {
         })
     }
 
-    udpSend(message) {
+    resolvHost(host) {
+        return new Promise((resolv, reject) => {
+            if(net.isIPv4(host)) resolv([host])
+            const resolver = new dns.Resolver()
+            resolver.setServers(["1.1.1.1", "8.8.8.8", "9.9.9.9"])
```
Comment (carlhoerberg) [inline src/sparoid.mjs]: No possible to use the locally configured DNS resolvers?

Comment (dentarg) [reply src/sparoid.mjs]: I didn't try, didn't think of it, added these seeing we use external servers in a few other projects.

what do we prefer?

**Comment by carlhoerberg** [comment]
I think where you've seen that is when we use opendns to resolve our public
IP using their special record for that. Would prefer using the local DNS
when possible.

**Comment by dentarg** [comment]
Sure

**Comment by dentarg** [comment]
@carlhoerberg see the commit message for c5e59a855e23d9cb013fac5a668dd3884cb26b5f, as stated, not sure what's going on there...

**Comment by dentarg** [comment]
Okay, I think https://github.com/84codes/sparoid.js/commit/0a3d0a722727d51d22f622298be6e47c5d0002a8 is a better version of this (no loop in `udpSend`) and I [shouldn't use `forEach`](https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop)

File: src/sparoid.mjs [hunk 2]
```diff
@@ -13,10 +14,14 @@ export default class Sparoid {
...
     async auth() {
+        const self = this
         const msg = await this.plainMsg()
         const encrypted = this.encrypt(msg)
         const hmaced = this.prefixHmac(encrypted)
-        await this.udpSend(hmaced)
+        const addresses = await this.resolvHost(self.host)
+        await Promise.all(addresses.map(async (address) => {
+            await this.udpSend(hmaced, address)
+        }));
```
Comment (carlhoerberg) [inline src/sparoid.mjs]: With `await` without `try/catch` rejections would be unhandled, I recommend: 

```suggestion
        await Promise.all(addresses.map((address) => self.udpSend(hmaced, address)));
```

Not sure if `self = this` is required anymore.

File: src/sparoid.mjs [hunk 3]
```diff
@@ -74,11 +79,23 @@ export default class Sparoid {
         })
     }
 
-    udpSend(message) {
+    resolvHost(host) {
+        return new Promise((resolv, reject) => {
+            if(net.isIPv4(host)) resolv([host])
```
Comment (carlhoerberg) [inline src/sparoid.mjs]:
```suggestion
            if (net.isIPv4(host)) resolv([host])
```

**Review by carlhoerberg — CHANGES_REQUESTED** [review]

File: src/sparoid.mjs (hunk 2 above)
Comment (dentarg) [reply src/sparoid.mjs]: > With `await` without `try/catch` rejections would be unhandled, I recommend

Thanks, I did that

> Not sure if `self = this` is required anymore.

It is, it is not working without it, we pass `self.host` as argument to `resolvHost()`

**Review by carlhoerberg — APPROVED** [review]

**Pull request merged**